assets/src/higher-order-components/with-latest-checkin.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External imports
+ */
+import { withSelect, withDispatch } from '@wordpress/data';
+import { createHigherOrderComponent, compose } from '@wordpress/compose';
+import { isModelEntityOfModel } from '@eventespresso/validators';
+
+/**
+ * Exposes properties related to the latest check-in entity for a given
+ * registration and datetime.
+ *
+ * Properties passed through to wrapped component are:
+ *
+ * - checkInEntity {BaseEntity|null} The check-in entity that is related to the
+ * given registration and datetime id.
+ * - hasResolvedCheckin {boolean} Whether the check-in entity selector has
+ * resolved.  This is important because its possible there is no entity for
+ * this registration and datetime if that registration hasn never been checked
+ * in.
+ * - onClick {function} A click handler which when invoked, will toggle the
+ * check-in state for the given registration and datetimeId. Note: this will
+ * replace the store latestCheckin record in the state for this given
+ * registration and datetime id which will get picked up by the `withSelect`
+ * HOC in the composed component.
+ *
+ * @type {WPComponent}
+ */
+const withLatestCheckin = createHigherOrderComponent(
+	compose( [
+		withSelect(
+			( select, { registration, datetimeId } ) => {
+				if ( ! isModelEntityOfModel(
+					registration,
+					'registration'
+				) ) {
+					return {};
```

Comment:
what if the registration is invalid? 
what about using a warning here so at least there will be some kind of indication what the problem is?

Reply:
I think that's something that should be left up to the wrapped component to do.  The messaging may differ on how a wrapped component handles when a registration isn't valid (wrapped components would still receive the `registration` prop).  This is predominately in place to short circuit the `withLatestCheckin` logic when the registration isn't valid because there's nothing it can do.

Reply:
I went to implement the warning anyways but then remembered again that I intentionally did things this way because there may be legitimate reasons for a parent component to pass through a `null` value for the registration (or when this hoc is composed with other hocs that retrieve a registration).